snippets/current/src/issue-a-credential.ts

```diff
@@ -1,215 +1,189 @@
+          connectOnStartup: true,
+        },
+      ],
+    }),
+    anoncreds: new AnonCredsModule({
+      registries: [new IndyVdrAnonCredsRegistry()],
+    }),
+    dids: new DidsModule({
+      resolvers: [new IndyVdrIndyDidResolver()],
+    }),
+    credentials: new CredentialsModule({
+      credentialProtocols: [
+        new V2CredentialProtocol({
+          credentialFormats: [new LegacyIndyCredentialFormatService(), new AnonCredsCredentialFormatService()],
+        }),
+      ],
+    }),
+  },
+})
+
+// Register a simple `WebSocket` outbound transport
+holder.registerOutboundTransport(new WsOutboundTransport())
+
+// Register a simple `Http` outbound transport
+holder.registerOutboundTransport(new HttpOutboundTransport())
+
+// Register a simple `Http` inbound transport
+holder.registerInboundTransport(new HttpInboundTransport({ port: 3002 }))
 // end-section-2
 
 // start-section-3
-// const registerSchema = async (issuer: Agent) =>
-//   issuer.ledger.registerSchema({ attributes: ['name', 'age'], name: 'Identity', version: '1.0' })
+holder.events.on<CredentialStateChangedEvent>(CredentialEventTypes.CredentialStateChanged, async ({ payload }) => {
+  switch (payload.credentialRecord.state) {
+    case CredentialState.OfferReceived:
+      console.log('received a credential')
+      // custom logic here
+      await holder.credentials.acceptOffer({ credentialRecordId: payload.credentialRecord.id })
+    case CredentialState.Done:
+      console.log(`Credential for credential id ${payload.credentialRecord.id} is accepted`)
+      // For demo purposes we exit the program here.
+      process.exit(0)
+  }
+})
 // end-section-3
 
 // start-section-4
-// const registerCredentialDefinition = async (issuer: Agent, schema: Schema) =>
-//   issuer.ledger.registerCredentialDefinition({ schema, supportRevocation: false, tag: 'default' })
+issuer.credentials.offerCredential({
+  protocolVersion: 'v2',
+  connectionId: '<connection id>',
+  credentialFormats: {
+    indy: {
+      credentialDefinitionId: '<credential definition id>',
+      attributes: [
+        { name: 'name', value: 'Jane Doe' },
+        { name: 'age', value: '23' },
+      ],
+    },
+  },
+})
 // end-section-4
 
 // start-section-5
-const setupCredentialListener = (holder: Agent) => {
-  holder.events.on<CredentialStateChangedEvent>(CredentialEventTypes.CredentialStateChanged, async ({ payload }) => {
-    switch (payload.credentialRecord.state) {
-      case CredentialState.OfferReceived:
-        console.log('received a credential')
-        // custom logic here
-        await holder.credentials.acceptOffer({ credentialRecordId: payload.credentialRecord.id })
-      case CredentialState.Done:
-        console.log(`Credential for credential id ${payload.credentialRecord.id} is accepted`)
-        // For demo purposes we exit the program here.
-        process.exit(0)
-    }
-  })
-}
+issuer.credentials.offerCredential({
```

Comment:
```suggestion
const credentialExchangeRecord = await issuer.credentials.offerCredential({
```

Reply:
both being called `const credentialExchangeRecord` won't work, butI think it's good to at least add an `await` (to show it's a promise)